src/server/server_family.cc

```diff
@@ -483,11 +484,15 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
   listeners_ = std::move(listeners);
   dfly_cmd_ = make_unique<DflyCmd>(this);
 
-  SetMaxClients(listeners_, absl::GetFlag(FLAGS_maxclients));
+  SetMaxClients(listeners, absl::GetFlag(FLAGS_maxclients),
```

Comment:
why here you are call it directly? how it is different from below?

Reply:
Here, it's part of the `init` process. The one below is used for `CONFIG SET`. The different is where it calls `SetMaxClients`. The former, just calls the member SetMaxClient for each listener. That later, runt it on the same Proactor as the one that runs the Listener. That way, the `read` and `write` part on the `maxclient` is synchronized. Also, the reason that we can't use the same chore (that is, call `listener->socket`) on the init, is because that `socket` is `nullptr` (it is initialized after `Init`) when we call `acceptor->AddListener(bind_addr, port, main_listener);`

Reply:
Ok, seems to me that `service.Init(acceptor, listeners, opts);` was arbitrary called before adding the main listener.
If I am mistaken and `Init` must be called before adding `main_listener` than we should add a comment there why it is the case.

Reply:
I agree. There is no reason why `Init` should be called before adding the listener so I addressed that and removed the template :) Imo, there are a lot of things I want to refactor in dfly_main but that's beyond the scope of this PR.